msgraphcore/middleware/options/middleware_control.py

```diff
@@ -7,3 +11,22 @@ def set(self, middleware_option_name, middleware_option):
 
     def get(self, middleware_option_name):
         return self.middleware_options.get(middleware_option_name, None)
+
+    def get_middleware_options(self, func):
+        self._reset_middleware_options()
+
+        def wrapper(*args, **kwargs):
+            # Get middleware options from **kwargs
+            scopes = kwargs.pop('scopes', None)
+            if scopes:
+                # Set middleware options, for use by middleware in the middleware pipeline
+                self.set(AUTH_MIDDLEWARE_OPTIONS, AuthMiddlewareOptions(scopes))
+            return func(*args, **kwargs)
+        return wrapper
+
+    def _reset_middleware_options(self):
+        # Reset middleware, so that they are not persisted across requests
+        self.middleware_options = {}
+
+
+middleware_control = MiddlewareControl()
```

Comment:
>MiddlewareControl [](start = 21, length = 17)

not sure i like this name since the class is really just about handling the middleware options dictionary but MiddlewareControl makes me think it's doing much more.

Reply:
I used the name for consistency with other sdks. If we are not concerned about consistency, I can rename it `MiddlewareOption`

Reply:
consitency argument wins.